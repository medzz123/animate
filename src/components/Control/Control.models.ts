export interface ControlProps {
  handleReset: () => void;
}
