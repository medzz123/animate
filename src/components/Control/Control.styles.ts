import styled from 'styled-components';

import { tokens } from '../../theme/tokens';

export const ControlContainer = styled.div`
  display: flex;
  justify-content: space-between;

  padding: ${tokens.sizes[8]};
  overflow-x: scroll;

  -ms-overflow-style: none; /* IE and Edge */
  scrollbar-width: none; /* Firefox */

  &::-webkit-scrollbar {
    display: none;
  }
`;

export const NewFrameInput = styled.input`
  border: none;
  box-shadow: inset 0 0 0 1px ${(p) => p.theme.paragraph};
  background-color: transparent;
  padding-left: 4px;
  width: 26px;
  color: ${(p) => p.theme.paragraph};
  height: 26px;

  &:focus {
    outline: none;
  }
`;
