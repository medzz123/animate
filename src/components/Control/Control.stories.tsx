import { Meta, Story } from '@storybook/react/types-6-0';
import React from 'react';

import Control from './Control';

export default {
  title: 'Components/Control',
  component: Control,
} as Meta;

const Template: Story = (storyArguments) => <Control {...storyArguments} />;

export const ControlStory = Template.bind({});
