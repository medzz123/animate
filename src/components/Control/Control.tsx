import React, { FunctionComponent } from 'react';
import { FaPause, FaPlay, FaStop } from 'react-icons/fa';

import Box from '../Box';
import Flex from '../Box/Flex';
import Button from '../Button';
import { ControlContainer } from './Control.styles';

const Control: FunctionComponent = () => {
  return (
    <ControlContainer data-testid="control">
      <Button>Add Frame</Button>
      <div>
        <Button>Duration</Button>
      </div>
      <Flex>
        <Button size="icon">
          <FaPlay />
        </Button>
        <Box width={8} />
        <Button size="icon">
          <FaPause />
        </Button>
        <Box width={8} />
        <Button size="icon">
          <FaStop />
        </Button>
      </Flex>
    </ControlContainer>
  );
};

export default Control;
