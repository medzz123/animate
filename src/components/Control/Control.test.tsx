import React from 'react';

import { render } from '../../../__tests__/testUtils';
import Control from './Control';

test('Control Test', () => {
  const { container } = render(<Control />);

  expect(container).toMatchSnapshot();
});
