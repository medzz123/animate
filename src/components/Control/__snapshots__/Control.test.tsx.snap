// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Control Test 1`] = `
<div>
  <div
    class="sc-dkPtyc ivLdOD"
    data-testid="control"
  >
    <button
      class="sc-gsDJrp liakXf"
      data-testid="button"
      type="button"
    >
      Add Frame
    </button>
    <div>
      <button
        class="sc-gsDJrp liakXf"
        data-testid="button"
        type="button"
      >
        Duration
      </button>
    </div>
    <div
      class="sc-bdvvaa hOlXtj"
      data-testid="box"
    >
      <button
        class="sc-gsDJrp bzHnWL"
        data-testid="button"
        type="button"
      >
        <svg
          fill="currentColor"
          height="18"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 448 512"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M424.4 214.7L72.4 6.6C43.8-10.3 0 6.1 0 47.9V464c0 37.5 40.7 60.1 72.4 41.3l352-208c31.4-18.5 31.5-64.1 0-82.6z"
          />
        </svg>
      </button>
      <div
        class="sc-bdvvaa kkNaPP"
        data-testid="box"
      />
      <button
        class="sc-gsDJrp bzHnWL"
        data-testid="button"
        type="button"
      >
        <svg
          fill="currentColor"
          height="18"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 448 512"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M144 479H48c-26.5 0-48-21.5-48-48V79c0-26.5 21.5-48 48-48h96c26.5 0 48 21.5 48 48v352c0 26.5-21.5 48-48 48zm304-48V79c0-26.5-21.5-48-48-48h-96c-26.5 0-48 21.5-48 48v352c0 26.5 21.5 48 48 48h96c26.5 0 48-21.5 48-48z"
          />
        </svg>
      </button>
      <div
        class="sc-bdvvaa kkNaPP"
        data-testid="box"
      />
      <button
        class="sc-gsDJrp bzHnWL"
        data-testid="button"
        type="button"
      >
        <svg
          fill="currentColor"
          height="18"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 448 512"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M400 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48z"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;
